wp-includes/js/service-worker-navigation-routing.js

```diff
@@ -8,7 +8,16 @@ wp.serviceWorker.routing.registerRoute( new wp.serviceWorker.routing.NavigationR
 
 		const handleResponse = ( response ) => {
 			if ( response.status < 500 ) {
-				return response;
+				if ( response.redirected ) {
+					const redirectedUrl = new URL( response.url );
+					redirectedUrl.searchParams.delete( STREAM_HEADER_FRAGMENT_QUERY_VAR );
+					const script = `<script>history.replaceState( {}, '', ${ JSON.stringify( redirectedUrl.toString() ) } );</script>`;
+					return response.text().then( ( body ) => {
+						return new Response( script + body );
```

Comment:
@jeffposnick here's how I ended up dealing with the challenge of handling a streamed body that redirects.

The “Try Redirect” nav menu item points to an old post URL that redirects, so you can see the URL go from "old-post" to "new post".

Reply:
https://streaming-westonruter.pantheonsite.io/

Reply:
😍